SQL AAD Setup (#110)
UCLH-Foundry/FlowEHR: README.md

CI deployment workflows are run in [Github environments](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment). These should
be created in a private repository created from this template repository.

> This step will create an AAP Application and Service Principal in the specified tenancy, and grant that service principal permissions in Azure and AAD needed for deployment. These are detailed below. 

Reviewer: ```suggestion
This step will create an AAD Application and Service Principal in the specified tenancy, and grant that service principal permissions in Azure and AAD needed for deployment. These are detailed below. 
```

> _NOTE_: The user following the steps below will need to be an `Owner` of the target Azure Subscription as well as a `Global Administrator` in AAD.

1. Open this repo in the dev container, and create the `config.yaml` file as outlined above.

2. Create the service principal with required AAD permissions: 

    ```bash
    make ci-auth
    ```

    _NOTE_: CI deployments require a service principal with access to deploy resources
    in the subscription, and the following permissions within the associated AAD tenancy:
    - `Application.ReadWrite.All`: Required to query the directory for the MSGraph app, and create applications used to administer the SQL Server.   
    - `AppRoleAssignment.ReadWrite.All`: Required to assign the following permissions to the System Managed Identity for SQL Server. 

    - Copy the block of JSON from the terminal for the next step.

2. Create and populate a GitHub environment

    Add an environment called `Infra-Test` with the following secrets:

>     - `AZURE_CREDENTIALS`: json containing the credentials of the service principal in the format. This should have been output for you in the correct format in the previous step:

Reviewer: ```suggestion
    - `AZURE_CREDENTIALS`: json containing the credentials of the service principal (outputted from the previous step) in the following format:
```
